### PR TITLE
ENH: Add a rolling mean function

### PR DESCRIPTION
**Pull request by WaqarSM**
Hello all, 

This is my first contribution to NumPy - please excuse any errors. :) 

In my works in physics and data science, I often find myself writing a 'Rolling Average'. 
So I wrote it into NumPy.  It returns an array of the average over a given range/window.

### CODE REVIEW

**Comment by grlee77** [comment]
Hi @WaqarSM,

Thank you for your interest in contributing. Ideas for new features should typically first be raised on the [mailing list](https://mail.python.org/mailman/listinfo/numpy-discussion).

In this case, I doubt the core devs here will want to add this function because you can use the existing `numpy.convolve` with `mode='valid'` to get the same behavior.

```Python
import numpy as np
size = 2 # size of the window for averaging
data = np.arange(1, 6)
np.convolve(data, np.ones(size)/size, mode='valid')
```

I think the convolve-based solution will have better numerical accuracy than using `cumsum` (see #9329 regarding cumulative floating point errors in `cumsum`)


If your averaging window size is somewhat large, I would consider SciPy's `uniform_filter1d` rather than `numpy.convolve` because it has a specialized implementation that can skip some redundant computations as the window moves.

```Python
import numpy as np
import scipy.ndimage as ndi

# apply running mean over a window of length 64 along axis 0 of data
size = 64
data = np.random.randn(10000)
rolling_avg = ndi.uniform_filter1d(data, size=size, axis=0)
```

File: numpy/__init__.pyi [hunk 1]
```diff
@@ -1235,3 +1235,7 @@ def amin(
     initial: _NumberLike = ...,
     where: _ArrayLikeBool = ...,
 ) -> Union[number, ndarray]: ...
+def rollingmean(
+    a: _ArrayLike,
```
Comment (charris) [inline numpy/__init__.pyi:1239]: Is the underscore intentional?

**Comment by charris** [comment]
Needs a release note.

**Comment by charris** [comment]
The  failing public API test also needs an update.

**Comment by mattip** [comment]
@WaqarSM do you wish to pursue this? If not please close it.

**Comment by WaqarSM** [comment]
> @WaqarSM do you wish to pursue this? If not please close it.

Yes, I do. I will address issues and email the mailing list regarding this feature this weekend. :) 

Thank you for all your helpful comments all.

File: numpy/__init__.pyi (hunk 1 above)
Comment (WaqarSM) [reply numpy/__init__.pyi:1239]: Oh! Good catch, it was unintentional. Will fix. :)

**Comment by WarrenWeckesser** [comment]
There have been no updates since the initial commits in July 2020, and @grlee77 has pointed out a couple alternatives that do not require a new function in NumPy.

Thanks @WaqarSM for the contribution, but at this point, it looks like this can be closed.

**Pull request closed without merging**